lptree.pas

```diff
@@ -5156,6 +5157,23 @@ function TLapeTree_Method.Compile(var Offset: Integer): TResVar;
     FCompiler.DecStackInfo(Offset, True, True, False, @_DocPos);
     FCompiler.Emitter._JmpR(Offset - fo, fo, @_DocPos);
   end;
+
+  if (FCompiler.HasHintDefine) then
+    with FStackInfo do
+      for i := 0 to VarCount - 1 do
+      begin
+        if (Vars[i].Used <> duFalse) or (Vars[i].Name = '') or (Vars[i].Name[1] = '!') then
+          Continue;
+
+        if (Vars[i] is TLapeParameterVar) then
+        begin
+          if (FStackInfo.Vars[i].Name = 'Result') then
```

Comment:
I think this might better checking for `out` parameters, rather than only `Result`.